change_detect.go

```diff
@@ -106,21 +106,19 @@ func (g *changeDetectGroup) DeleteGroup(d edge.DeleteGroupMessage) (edge.Message
 }
 func (g *changeDetectGroup) Done() {}
 
-// changeDetect calculates the changeDetect between prev and cur.
-// Return is the resulting changeDetect, whether the current point should be
-// stored as previous, and whether the point result should be emitted.
+// changeDetect reports whether there was a change between prev and cur.
 func (n *ChangeDetectNode) changeDetect(prev, curr models.Fields) bool {
-
-	value, ok := curr[n.d.Field]
-	if !ok {
-		n.diag.Error("Invalid field in change detect",
-			fmt.Errorf("expected field %s not found", n.d.Field),
-			keyvalue.KV("field", n.d.Field))
-		return false
-	}
-	if prev[n.d.Field] == value {
-		return false
+	for _, field := range n.d.Fields {
+		value, ok := curr[field]
+		if !ok {
+			n.diag.Error("Invalid field in change detect",
```

Comment:
Do you think that this might be a bit noisy? I'm imagining a case where you want to change detect on `field_a` or `field_b`, but 90% of the contains data that is written is for only `field_a`. Wont this create a log for every point that comes through (provided that there is no change detected each time?

Reply:
Good question. All nodes now have the `.quiet` option to tell them to not log anything. So I think it if fine to leave as the default then in that special case the user can explicitly quite the specific changeDetect node.